canonical-web-and-design/maas-ui
# feat(ui): build UploadTextArea component

## Done

- Built `UploadTextArea` component which combines a file input and textarea
- Used in Add/Edit KVM forms and replaced `UserDataField` which shares about 80% of the code

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click "Add KVM"
- Select the "Provide certificate and key" radio
- Check that you can manually type in the textarea, click the "Upload" button or drop a file into the textarea
- Go to the machine list, select a Ready machine and choose "Deploy" from the action dropdown
- Select "Cloud-init user data" and check that the upload component works here as in the Add KVM form

## Fixes

Fixes canonical-web-and-design/app-squad#244

## Screenshot
![Peek 2021-09-21 11-05](https://user-images.githubusercontent.com/25733845/134097295-f885100c-4af9-4599-9c02-8414345e1132.gif)
